lua/harpoon/ui.lua

```diff
@@ -65,8 +64,16 @@ M.toggle_quick_menu = function()
         contents[idx] = string.format("%d %s", idx, file)
     end
 
+    vim.api.nvim_buf_set_name(bufh, "harpoon-menu")
     vim.api.nvim_buf_set_lines(bufh, 0, #contents, false, contents)
     vim.api.nvim_buf_set_option(bufh, "filetype", "harpoon")
+    vim.api.nvim_buf_set_option(bufh, "buftype", "acwrite")
+    vim.cmd(string.format("autocmd BufWriteCmd <buffer=%s> :lua require('harpoon.ui').on_menu_save()", bufh))
+end
+
+M.on_menu_save = function()
+    save_changes()
+    vim.cmd("set nomodified")
```

Comment:
Interesting, so this is the piece I was missing?

Reply:
If modified is not reset, you indicate the write was *not* successful, + you get error when closing nvim that you have modified buffers.

You were very close though, I think your missing piece was naming the buffer with `vim.api.nvim_buf_set_name`.